ukis_pysat/data.py

```diff
@@ -72,75 +90,75 @@ def __init__(self, datahub, datadir=None, datadir_substr=None):
-            end_date = sentinelsat.format_query_date(date[1])
+        if self.src == Datahub.Directory:
+            # query STAC Catalog for metadata
+            catalog = self._init_catalog()
             geom = self.prep_aoi(aoi)
-            if cloud_cover and platform != platform.Sentinel1:
-                min_cloud_cover = cloud_cover[0]
-                max_cloud_cover = cloud_cover[1]
-            else:
-                min_cloud_cover = 0
-                max_cloud_cover = 100
-
-            # get all json files in datadir that match substr
-            meta_files = sorted(
-                [
-                    Path(dp).joinpath(f)
-                    for dp, dn, filenames in os.walk(self.api)
-                    for substr in self.api_substr
-                    for f in filenames
-                    if f.endswith(".json") and substr in f
-                ],
-                key=lambda path: str(path).lower(),
-            )
-
-            # filter json files by query parameters
-            meta_src = []
-            for meta_file in meta_files:
-                with open(meta_file) as f:
-                    m = json.load(f)
-                    try:
-                        self.construct_metadata(meta_src=m, platform=platform)
-                    except (json.decoder.JSONDecodeError, LookupError, TypeError) as e:
-                        raise ValueError(f"{meta_file.name} not a valid metadata file. {e}.")
-                    m_platform = m["properties"]["platformname"]
-                    m_date = sentinelsat.format_query_date(m["properties"]["acquisitiondate"])
-                    m_geom = geometry.shape(m["geometry"])
-                    m_cloud_cover = m["properties"]["cloudcoverpercentage"]
-                    if m_cloud_cover is None:
-                        m_cloud_cover = 0
-                    if (
-                        m_platform == platform.value
-                        and start_date <= m_date < end_date
-                        and m_geom.intersects(geom)
-                        and min_cloud_cover <= m_cloud_cover < max_cloud_cover
-                    ):
-                        meta_src.append(m)
+            for item in self.api.get_all_items():
+                if item.ext.eo.cloud_cover and cloud_cover:
+                    if not cloud_cover[0] <= item.ext.eo.cloud_cover < cloud_cover[1]:
+                        continue
+                if (
+                    platform.value == item.common_metadata.platform
+                    and sentinelsat.format_query_date(date[0])
+                    <= sentinelsat.format_query_date(parse(item.properties["acquisitiondate"]).strftime("%Y%m%d"))
+                    < sentinelsat.format_query_date(date[1])
+                    and geometry.shape(item.geometry).intersects(geom)
+                ):
+                    catalog.add_item(item)
+            return catalog
+
+        elif self.src == Datahub.StacApi:

```

Comment:
If we do not expose here the query method, why do we actually add StacAPI to Source in the first place? As far as I can see, we either remove StacApi completely from Source class and refer to StacApi class directly OR we expose here the same query interface as for the other sources (knowing that it does not capture yet all the flexibility of StacAPI).

Reply:
using kwargs like with the other sources we could even allow full query potential for StacApi.

Reply:
I suggest to pick this up in #103